File: front_end/lib/dto/auto_cadastro/authResponsedto.dart
import 'package:crud_flutter/dto/auto_cadastro/user_response_dto.dart';

class AuthResponseDTO {
  final UserResponseDTO usuario;
  final String token;

  AuthResponseDTO({
    required this.usuario,
    required this.token,
  });

  factory AuthResponseDTO.fromJson(Map<String, dynamic> json) {
    return AuthResponseDTO(
      usuario: UserResponseDTO.fromJson(json['usuario']),
      token: json['token'],
    );
  }
}

File: back-end/src/main/java/com/tnl/listacompras/dto/responseDTO/auto_cadastro/AuthResponseDTO.java
package com.tnl.listacompras.dto.responseDTO.auto_cadastro;

public class AuthResponseDTO {

    private UsuarioResponseDTO usuario;
    private String token;

    public AuthResponseDTO(UsuarioResponseDTO usuario, String token) {
        this.usuario = usuario;
        this.token = token;
    }

    public UsuarioResponseDTO getUsuario() {
        return usuario;
    }

    public String getToken() {
        return token;
    }
}